### PR TITLE
EAFT

### PR DESCRIPTION
**Pull request by salmanmohammadi**
WandB logs here https://wandb.ai/axolotl-ai/eaf

## Summary by CodeRabbit

* **New Features**
  * Added EAFT (Efficient Adapters for Fine-Tuning) training support with entropy-aware focal loss computation for improved model training.
  * Introduced new configuration parameters: use_eaft (toggle feature), eaft_alpha (entropy weighting exponent), and eaft_k (number of top logits).
  * Included example EAFT training configuration file demonstrating all settings and usage options.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by winglian — APPROVED** [review]
Lgtm, might be good to follow this up with an efficient chunked cross entropy version later.

File: src/axolotl/monkeypatch/loss/eaft.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    mask = shift_labels_view != -100
+
+    with torch.no_grad():
+        top_k_logits, _ = torch.topk(
+            shift_logits_view[mask].float(), k=min(k, vocab_size), dim=-1
+        )
+        top_k_probs = F.softmax(top_k_logits, dim=-1)
+        entropy = -(top_k_probs * torch.log(top_k_probs + 1e-10)).sum(dim=-1)
+        weights = torch.pow(entropy, alpha)
```
Comment (leeparkuky) [inline src/axolotl/monkeypatch/loss/eaft.py:40]: Entropy should be divided by 3 (ln20) before torch.pow
